include/GIL/Module.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef GLU_GIL_MODULE_HPP_
+#define GLU_GIL_MODULE_HPP_
+
+#include <list>
+#include <string>
+#include <memory>
+#include "Function.hpp"
+
+namespace glu::gil {
+
+class Type;
+
+/// @brief This class represents modules in our gil intermediate representation
+/// Modules are used to automaticaly import all delcarations in a file,
+/// see https://glu-lang.org/modules/
+class Module {
+public:
+    /// @brief Constructor for a Module instance that initialise its name
+    /// @param moduleName A string representing the modules name
+    Module(std::string moduleName);
+
+    /// @brief Adds a Function to the Functions dedicated chained list
+    /// @param function A pointer to the function that will be added
+    void addFunction(Function &&function);
```

Comment:
there's two possibilities:
- you take a unique_ptr so that the function doesn't change its address
- you create the function in place (emplace)
functions must have a stable address, we keep pointers to them all the time. they should have a deleted move ctor

Reply:
for now I will do the unique pointer method and might add the emplace later after Function class is done

Reply:
I think emplacing makes more sense, we don't want Functions that aren't inside a module

Reply:
the best would be if Function's ctor is private, so they can only be created by modules

Reply:
I agree but we'll have to wait for the functions PR to be done I close this PR for now